### ISSUE TITLE
Page stuck on environment checking indefinitely when call to get secure mode fails

### ISSUE THREAD

**Issue by ajcasagrande** [edgexfoundry/edgex-ui-go#536]
Not a big deal, but an observation I made.

# 🐞 Bug Report

### Affected Services [**REQUIRED**]
edgex-ui

### Is this a regression?
Not sure

### Description and Minimal Reproduction [**REQUIRED**]
Run `ng serve`, but do not run the go binary. Load `localhost:4000` (or whatever you serverd it on) and the page will fail to proxy getSecureMode. You will end up with a stuck page spinning the "environment checking" indefinitely.

The UI should probably handle the error case from that call and redirect to an error page.

## 🌍  Your Environment
**Deployment Environment:**

**EdgeX Version [**REQUIRED**]:**

**Anything else relevant?**

**Comment by badboy-huaqiao** [comment]
@ajcasagrande yeah, good catch, but that's will never happen in product environment, the development env is different from the deployment environment(or let's call it product environment), the GUI backend server must be running if you want to develop it.

- in development environment, both ng serve and GUI backend server are required
- in product/deployment environment, only GUI backend server is required
